src/Proxy.hs

```diff
@@ -890,5 +894,36 @@ putGTUDropR addrText = do
 getGlobalFileR :: Handler TypedContent
 getGlobalFileR = toTypedContent . globalInfo <$> getYesod
 
+-- Queries the transaction database and the GRPC, 
+-- then if both succeed checks that the last final block is less than 5 minutes old.
+getHealthR :: Handler TypedContent
+getHealthR = 
+  runGRPC doGetBlockFinalInfo $ \case
+      Nothing -> do
+        $(logError) $ "Could not get response from GRPC."
+        sendResponse $ object $ ["healthy" .= False, "reason" .= ("Could not get response from GRPC.":: String)]
+      Just lastFinalBlockInfo -> do
+        $(logInfo) "Successfully got best block info."
+        _ :: [(Entity Entry, Entity Summary)] <- runDB $ E.select $
+                  E.from $ \val -> do
+                  E.limit 1
+                  return val
+        -- get block slot time from block info object, compare with current time: reject if more than 5 minutes
+        case fromJSON lastFinalBlockInfo of
+          Error _ -> sendResponse $ object $ ["healthy" .= False, "reason" .= ("Block info format has changed. blockSlotTime is not UTCTime.":: String)]
+          Success (bir :: BlockInfoResult) -> do
+            currentTime <- liftIO Clock.getCurrentTime
+            case (Clock.diffUTCTime currentTime (birBlockSlotTime bir)) < (Clock.secondsToNominalDiffTime 300) of -- 5 minutes = 300 seconds
```

Comment:
It would be better if 300 is a parameter given when the proxy is started.

300 is a good default value, but it should be easy to change.